vowpalwabbit/gd.cc

```diff
@@ -479,7 +480,11 @@ float get_pred_per_update(gd& g, example& ec)
 { //We must traverse the features in _precisely_ the same order as during training.
   label_data& ld = ec.l.simple;
   vw& all = *g.all;
-  float grad_squared = all.loss->getSquareGrad(ec.pred.scalar, ld.label) * ec.weight;
+
+  float grad_squared = ec.weight;
+  if (count(all.args.begin(), all.args.end(),"--cs_active") == 0)
```

Comment:
Another strange test here.  Why does the estimate need to change for cs_active?

Reply:
I believe I know the answer to this. We decided that for cs_active, it is better to use sum of x^2 rather than gradient squared as the metric for updating the weights and confidence bounds. The use of gradient squared driven metric in computing sensitivities was causing a lot of issues. Once your learner becomes good, it makes few mistakes in prediction so that the gradients are small, and hence the squared gradients do not grow. That ends up meaning that your confidence plateaus and you maintain a reasonable chance of querying things. This does not happen when things are based on squared x values, which also corresponds better to previous works with linear representations specifically.

Reply:
I'm not quite understanding.  getSquareGrad gets the squared gradient, right?

Reply:
Yes, but you only need the squared gradient as a multiplier for adagrad, which we do not want in cs_active.